extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -211,7 +212,9 @@ void addAdditionalRoutes(
             BuildProducer<RouteBuildItem> routeProducer,
             BuildProducer<FilterBuildItem> filterProducer,
             List<RequireBodyHandlerBuildItem> bodyHandlerRequired,
-            BeanArchiveIndexBuildItem beanArchive) throws IOException {
+            BeanArchiveIndexBuildItem beanArchive,
+            // Make sure the routes are produced after the CDI container is ready - we need to initialize some state in the generated RouteHandler constructors
+            BeanContainerBuildItem beanContainer) throws IOException {
```

Comment:
I think you also want to produce a ServiceStartBuildItem to make sure that this is ready when the HTTP port is opened.

Reply:
> This is RUNTIME_INIT, isn't the bean container started in STATIC_INIT so will always be ready?

Ah yes, that's a good point.

> I think you also want to produce a ServiceStartBuildItem to make sure that this is ready when the HTTP port is opened.

That shouldn't be necessary because this build step produces `RouteBuildItem` which is consumed by a [build step](https://github.com/quarkusio/quarkus/blob/master/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java#L89-L103) that is a dependant of a [build step](https://github.com/quarkusio/quarkus/blob/master/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java#L111-L156) that produces `ServiceStartBuildItem("vertx-http")`.